test/facility/get-facility-by-facility-identifier.api-test.ts

```diff
@@ -0,0 +1,121 @@
+import { PROPERTIES } from '@ukef/constants';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { withAcbsAuthenticationApiTests } from '@ukef-test/common-tests/acbs-authentication-api-tests';
+import { IncorrectAuthArg, withClientAuthenticationTests } from '@ukef-test/common-tests/client-authentication-api-tests';
+import { Api } from '@ukef-test/support/api';
+import { ENVIRONMENT_VARIABLES, TIME_EXCEEDING_ACBS_TIMEOUT } from '@ukef-test/support/environment-variables';
+import { GetFacilityGenerator } from '@ukef-test/support/generator/get-facility-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import nock from 'nock';
+
+describe('GET /facilities/{facilityIdentifier}', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+  const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+  const facilityIdentifier = valueGenerator.stringOfNumericCharacters({ length: 10 });
+  const getFacilityUrl = `/api/v1/facilities/${facilityIdentifier}`;
+
+  const { facilitiesInAcbs, facilitiesFromApi } = new GetFacilityGenerator(valueGenerator, dateStringTransformations).generate({
+    numberToGenerate: 1,
+    facilityIdentifier,
+    portfolioIdentifier,
+  });
+
+  const facilityInAcbs = facilitiesInAcbs[0];
+  const expectedFacility = facilitiesFromApi[0];
+
+  let api: Api;
+
+  beforeAll(async () => {
+    api = await Api.create();
+  });
+
+  afterAll(async () => {
+    await api.destroy();
+  });
+
+  afterEach(() => {
+    nock.abortPendingRequests();
+    nock.cleanAll();
+  });
+
+  const { idToken, givenAuthenticationWithTheIdpSucceeds } = withAcbsAuthenticationApiTests({
+    givenRequestWouldOtherwiseSucceed: () => requestToGetFacility().reply(200, facilityInAcbs),
+    makeRequest: () => api.get(getFacilityUrl),
+  });
+
+  withClientAuthenticationTests({
+    givenTheRequestWouldOtherwiseSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      requestToGetFacility().reply(200, facilityInAcbs);
+    },
+    makeRequestWithoutAuth: (incorrectAuth?: IncorrectAuthArg) => api.getWithoutAuth(getFacilityUrl, incorrectAuth?.headerName, incorrectAuth?.headerValue),
+  });
+
+  it('returns a 200 response with the facility if it is returned by ACBS', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToGetFacility().reply(200, facilityInAcbs);
+
+    const { status, body } = await api.get(getFacilityUrl);
+
+    expect(status).toBe(200);
+    expect(body).toStrictEqual(JSON.parse(JSON.stringify(expectedFacility)));
+  });
```

Comment:
Can we also have API tests that check:
- `guaranteePercentage` is set to a default if `CompBalPctReserve` is null,
- `foreCastPercentage` is set to a default if `CompBalPctAmount` is null,

If we know any of the datetime fields could be null as well, then it'd be good to test their behaviour too

Reply:
Added API tests that check for the above and also null checks for all the datetime fields.